src/components/ComplaintLandingPage/AskIsUserOfApp.astro

```diff
@@ -3,31 +3,33 @@ import { t } from '../../i18n/server';
 import Base from '../../layouts/base.astro';
 import { absUrl } from '../../lib/util';
 import TextRaw from '../TextRaw.astro';
+import RadioForm from '../RadioForm.astro';
 
 interface Props {
     token: string;
+    complainantIsUserOfApp: boolean | null;
 }
 
-const { token } = Astro.props;
+const { token, complainantIsUserOfApp } = Astro.props;
+const isChecked = (p: string) =>
+    (complainantIsUserOfApp === true && p === 'yes') || (complainantIsUserOfApp === false && p === 'no');
 ---
 
 <Base title={t('complaint-landing-askIsUserOfApp', 'heading')}>
     <p>
         <TextRaw scope="complaint-landing-askIsUserOfApp" id="question" />
     </p>
 
-    <div class="col66 col100-mobile">
-        <div class="radio-group radio-group-vertical radio-group-padded">
-            {
-                (['yes', 'no'] as const).map((p) => (
-                    <form class="radio-wrapper" method="POST" action={absUrl(`/p/${token}/complain/askIsUserOfApp`)}>
-                        <input type="hidden" name="answer" value={p} />
-                        <button type="submit" class="radio-label">
-                            {t('common', p)}
-                        </button>
-                    </form>
-                ))
-            }
-        </div>
-    </div>
+    <RadioForm
+        actionUrl={absUrl(`/p/${token}/complain/askIsUserOfApp/answer`)}
+        options={(['yes', 'no'] as const).reduce(
+            (agg, p) =>
+                Object.assign(agg, {
+                    [p]: t('common', p),
+                }),
+            {},
+        )}
```

Comment:
I mean… I love a good `reduce` as much as the next person, but for two _very_ simple properties it's a bit silly (it's literally more code than just writing out the object! :D) and obfuscates the semantics.